src/pages/studyView/charts/violinPlotTable/StudyViewViolinPlot.tsx

```diff
@@ -0,0 +1,193 @@
+import * as React from 'react';
+import { observer } from 'mobx-react';
+import {
+    ClinicalViolinPlotIndividualPoint,
+    ClinicalViolinPlotBoxData,
+} from 'cbioportal-ts-api-client';
+import { computed } from 'mobx';
+import {
+    getViolinX,
+    violinPlotSvgHeight,
+    violinPlotXPadding,
+} from 'pages/studyView/charts/violinPlotTable/StudyViewViolinPlotUtils';
+
+export interface IStudyViewViolinPlotProps {
+    curveMagnitudes: number[];
+    violinBounds: { min: number; max: number };
+    individualPoints: ClinicalViolinPlotIndividualPoint[];
+    boxData: ClinicalViolinPlotBoxData;
+    showViolin: boolean;
+    showBox: boolean;
+    width: number;
+    gridValues: number[];
+}
+
+export const yPadding = 3;
+export const height = violinPlotSvgHeight - 2 * yPadding;
+
+@observer
+export default class StudyViewViolinPlot extends React.Component<
+    IStudyViewViolinPlotProps,
+    {}
+> {
+    @computed get svgWidth() {
+        return this.props.width;
+    }
+    @computed get violinWidth() {
+        return this.svgWidth - 2 * violinPlotXPadding;
+    }
+    renderCurve() {
+        if (this.props.curveMagnitudes.length > 0) {
+            const center = violinPlotSvgHeight / 2;
+
+            const curveLimit = height / 2;
+            const scale = Math.min(
+                1,
+                curveLimit / Math.max(...this.props.curveMagnitudes)
+            );
+            const curve = this.props.curveMagnitudes.map(y => scale * y);
+
+            const stepSize = this.violinWidth / (curve.length - 1);
+
+            let dUpperHalf = `M 0 ${center} L 0 ${center - curve[0]}`;
```

Comment:
maybe better comment here explaining what these are.  the violin curve is horizontal right?    not sure what "d" denotes and my SVG knowledge is pretty minimal so a little help would be good.